pkg/discovery/monitoring/master/cluster_monitor.go

```
@@ -107,22 +96,20 @@ func (m *ClusterMonitor) findClusterID() error {
 
 // ----------------------------------------- Node State --------------------------------------------
 func (m *ClusterMonitor) findNodeStates() {
-	for _, node := range m.nodeList {
-		key := util.NodeKeyFunc(node)
-		if key == "" {
-			glog.Warning("Invalid node")
-			continue
-		}
-		// node/pod/container cpu/mem resource capacities
-		m.genNodeResourceMetrics(node, key)
+	key := util.NodeKeyFunc(m.node)
+	if key == "" {
+		glog.Warning("Invalid node")
+		return
```

Should you return an error to caller?

---

Updated